official/recommendation/ncf_keras_benchmark.py

```diff
@@ -156,45 +176,69 @@ def benchmark_2_gpus_early_stop(self):
     FLAGS.num_gpus = 2
     self._run_and_report_benchmark()
 
-  # NCF with custom training loop. Works only in TF 2.0
   def benchmark_2_gpus_ctl(self):
+    """NCF with custom training loop. Works only in TF 2.0."""
     self._setup()
     FLAGS.keras_use_ctl = True
     FLAGS.num_gpus = 2
     self._run_and_report_benchmark()
 
-  # NCF with custom training loop. Works only in TF 2.0
   def benchmark_2_gpus_ctl_early_stop(self):
+    """NCF with custom training loop. Works only in TF 2.0."""
     self._setup()
     FLAGS.keras_use_ctl = True
     FLAGS.early_stopping = True
     FLAGS.num_gpus = 2
     self._run_and_report_benchmark()
 
   def benchmark_1_gpu_ctl_mlperf_like(self):
-    """1-GPU test to compare Google implementation with MLperf0.5.
-       Using similar rules as MLPerf0.5
-       Using Google's convergence hparams as base for 1-GPU test.
-       Fixed the number of epochs to 7, to remove the perf variance.
-       MLPerf submission consistently converges in 7 epochs.
+    """1-GPU test to compare Google implementation with MLPerf 0.5.
+
+       Using similar rules as MLPerf 0.5
+       - Using Google's convergence hparams as base for 1-GPU test.
+       - Fixed the number of epochs to 7, to remove the perf variance.
+       - MLPerf submission consistently converges in 7 epochs.
     """
     self._setup()
     FLAGS.keras_use_ctl = True
     FLAGS.train_epochs = 7
     self._run_and_report_benchmark()
 
   def benchmark_1_gpu_mlperf_like(self):
-    """1-GPU MLPerf like test with compile/fit version"""
+    """1-GPU MLPerf like test with compile/fit version."""
     self._setup()
     FLAGS.train_epochs = 7
     self._run_and_report_benchmark()
 
+  def benchmark_1_gpu_no_dist_strat_mlperf_like(self):
+    """1-GPU MLPerf like test with compile/fit version without dist_strat."""
+    self._setup()
+    FLAGS.train_epochs = 7
+    FLAGS.distribution_strategy = 'off'
+    self._run_and_report_benchmark()
+
+  def benchmark_1_gpu_no_dist_strat_run_eagerly_mlperf_like(self):
+    self._setup()
+    FLAGS.train_epochs = 7
+    FLAGS.distribution_strategy = 'off'
+    FLAGS.early_stopping = True
```

Comment:
i think you dont want early stopping in the mlperf like test iirc?

Reply:
You rock, saving me some big time pain later.  All fixed.